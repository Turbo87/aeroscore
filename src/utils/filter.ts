import fs = require('fs');

import {csvParse} from 'd3-dsv';

export function readHandicapsFromFile(path: string) {
  let content = fs.readFileSync(path, 'utf8');
  let lines = csvParse(content);

  let handicaps = Object.create(null);
  lines.forEach(({ CN, HANDICAP }) => {
    if (CN) {
      handicaps[CN] = HANDICAP ? parseInt(HANDICAP, 10) : 100;
    }
  });
  return handicaps;
}
