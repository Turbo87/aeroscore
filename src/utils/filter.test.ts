import * as filter from './filter';

describe('readHandicapsFromFile()', () => {
  let FIXTURES = [
    '2017-07-15-lev',
    'Bayreuth/2018-05-30/Open',
  ];

  for (let fixture of FIXTURES) {
    it(fixture, () => {
      let result = filter.readHandicapsFromFile(`${__dirname}/../../fixtures/${fixture}/filter.csv`);
      expect(result).toMatchSnapshot();
    });
  }
});
