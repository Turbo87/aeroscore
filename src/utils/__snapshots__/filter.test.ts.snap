// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`readHandicapsFromFile() 2017-07-15-lev 1`] = `
Object {
  "1": 101,
  "1G": 101,
  "1R": 104,
  "33": 98,
  "999": 100,
  "E1": 104,
  "EI": 107,
  "FR": 98,
  "H1": 105,
  "IGP": 104,
  "JM": 101,
  "LC": 104,
  "MI": 104,
  "MS": 98,
  "P9": 104,
  "SQ": 107,
  "SW": 98,
  "T": 104,
  "T1": 104,
  "TA": 101,
  "WJ": 104,
  "XN": 98,
  "ZG": 104,
}
`;

exports[`readHandicapsFromFile() Bayreuth/2018-05-30/Open 1`] = `
Object {
  "AM": 100,
  "DF": 100,
  "EPI": 100,
  "ET": 100,
  "FJ": 100,
  "FO": 100,
  "FU": 100,
  "HS": 100,
  "HW": 100,
  "I": 100,
  "IM": 100,
  "JOY": 100,
  "KV": 100,
  "LM": 100,
  "LY": 100,
  "MD": 100,
  "MM": 100,
  "NX": 100,
  "NY": 100,
  "QAX": 100,
  "RB": 100,
  "TEX": 100,
  "TOY": 100,
  "VH": 100,
  "VX": 100,
  "VY": 100,
  "W1": 100,
  "WR": 100,
  "YY": 100,
}
`;
